### ISSUE TITLE
Simplify getting JSON overrides into the SDKs

### ISSUE THREAD

**Issue by jdorn** [growthbook/growthbook#167]
It's really hard to follow best practices for our SDKs right now.

For the experiment overrides API endpoint, we recommend using Webhooks to keep a cached copy up to date. This requires creating a lambda function and writing code to insert to a database or cache and then reading from that cache to pass to the SDK and implementing local caching, timeout, and refresh logic.

We can greatly simplify this with two changes:
1.  Add a built-in S3 webhook option.  Give us the S3 bucket/path name and write access and we'll keep a JSON file up-to-date automatically.  This can replace a custom webhook, lambda, and a db/cache.
2.  Add a `configEndpoint` option to the SDK.  It takes the url of a JSON file, fetches it, and deals with local caching and timeout/refresh logic automatically.

These two changes would make the developer experience much smoother and encourage best practices.

As a first pass, we'll probably just do this for the javascript/react SDKs with others coming soon.

**Comment by jeroenvisser101** [comment]
This will be an optional thing, right? The current way of configuring may not be great for some, but being able to control the way configuration is fetched is vital for us. We include the A-B testing configuration from the initial response, so that SSR and CSR both have the same experiments running.

**Comment by jdorn** [comment]
Yes, if we do this it will definitely be optional. Either pass in a JSON object directly or specify an API key and endpoint and the SDK will deal with fetching and caching.
